fix(abs): open existing monero wallet if it already exists

Einliterflasche opened the pull request:
This PR introduces logic that, if we fail to generate a new monero wallet because it already exists, tries to open the existing wallet instead of aborting instantly.

Also ran dprint and clippy.

binarybaron commented:
I think there's already some other code that does the same. Can you check how it's done there (and fix it there as well if needed)

Einliterflasche commented:
I removed code from one instance where we did the fallback as well and now use the updated function. Also changed some function names to be more consistent.

binarybaron commented on swap/src/monero/wallet.rs:
```diff
@@ -26,33 +26,40 @@ pub struct Wallet {
 
 impl Wallet {
     /// Connect to a wallet RPC and load the given wallet by name.
-    pub async fn open_or_create(url: Url, name: String, env_config: Config) -> Result<Self> {
```
I think the previous name was better. It should be made explicit that we'll be creating a wallet if it does not exist.
